modules/electromagnetics/include/utils/ElectromagneticEnums.h

```diff
@@ -0,0 +1,32 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+/**
+ *  ElectromagneticEnums contains various enumerations useful in the
+ *  Electromagnetic module, such as real/imag component definitions in
+ *  Kernels, BCs, etc.
+ */
+namespace electromagnetics
+{
+/// Enum used when determining the component of a field vector or complex coefficient
+enum ComponentEnum
+{
+  REAL,
+  IMAGINARY
+};
+
+/// Enum used when determining the function of a Robin-style boundary condition
+enum RobinEnum
+{
+  ABSORBING,
+  PORT
```

Comment:
could use a docstring for each for the non-initiated

Reply:
Don't think I can use a docstring in a standard C++ enum looking at the std reference. Do you mean a better doxygen formatted comment about these?

Reply:
Ah I gotcha - will do. Here, a port in an electromagnetics simulation can be considered an entry point (for launching an electromagnetic wave), but also an exit point (for absorbing incoming waves launched elsewhere as well as any reflections of any field quantities that pass through it). The absorbing portion of these is like a first order approximation of the Sommerfeld radiation condition for truncating a simulation space. The portion strictly related to the wave launching describes how an incoming field with known components/structure interacts with the simulation at the boundary. Initially in a transient simulation (or with a simulation with very little reflection back to the port), you could have the literal incoming wave being emitted from the port (kind of like a Dirichlet condition in terms of the solution). The ABSORBING case here is, then, merely a PORT without an incoming wave. This might not be the case always, but given the first order versions of these I outline in `[Vector]EMRobinBC` they are additive in the integrated BC contribution.